src/formatDate.js

```diff
@@ -50,7 +50,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const DELIM = fromFormat.pop();
+  const GLUE = toFormat.pop();
+
+  const PREVIOUS_DATE_POSITION = fromFormat.indexOf('DD');
+  const NEXT_DATE_POSITION = toFormat.indexOf('DD');
+  const PREVIOUS_MONTH_POSITION = fromFormat.indexOf('MM');
+  const NEXT_MONTH_POSITION = toFormat.indexOf('MM');
+  const PREVIOUS_YEAR_POSITION = 3 - PREVIOUS_DATE_POSITION
+    - PREVIOUS_MONTH_POSITION;
+  const NEXT_YEAR_POSITION = 3 - NEXT_DATE_POSITION - NEXT_MONTH_POSITION;
+
+  const PERIOD_NUMBERS = date.split(DELIM);
+
+  const DATE_NUMBER = PERIOD_NUMBERS[PREVIOUS_DATE_POSITION];
+  const MONTH_NUMBER = PERIOD_NUMBERS[PREVIOUS_MONTH_POSITION];
```

Comment:
Please, fix namings. 

- Uppercase constants are typically used for values that are not expected to change throughout the execution of the program. By convention, they are often declared using uppercase letters and underscores to separate words.
- Uppercase constants are commonly used for configuration values, fixed values, or any other data that remains constant throughout the application's lifespan.


- Lowercase constants, like variables, are typically used for values that can change during the execution of the program but are not intended to be reassigned once initialized.
- Lowercase constants are suitable for storing values that may be calculated or determined at runtime, but remain constant once assigned.